Drop support for Python 3.4

Python 3.4 support pytest only versions < 3.3. Last added attribute `allow_module_level` for skipping is supported in pytest == 3.3.0, so tests against Python 3.4 broken:

```
=================================== ERRORS ====================================
___________________ ERROR collecting tests/test_fsevents.py ___________________
tests\test_fsevents.py:7: in <module>
    pytest.skip("macOS only.", allow_module_level=True)
E   TypeError: skip() got an unexpected keyword argument 'allow_module_level'
________________ ERROR collecting tests/test_inotify_buffer.py ________________
tests\test_inotify_buffer.py:23: in <module>
    pytest.skip("GNU/Linux only.", allow_module_level=True)
E   TypeError: skip() got an unexpected keyword argument 'allow_module_level'
__________________ ERROR collecting tests/test_inotify_c.py ___________________
tests\test_inotify_c.py:7: in <module>
    pytest.skip("GNU/Linux only.", allow_module_level=True)
E   TypeError: skip() got an unexpected keyword argument 'allow_module_level'
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!
=========================== 3 error in 1.08 seconds ===========================
ERROR: InvocationError for command 'C:\\projects\\watchdog\\.tox\\py34\\Scripts\\python.EXE -bb -m pytest' (exited with code 2)
___________________________________ summary ___________________________________
ERROR:   py34: commands failed
```

How about drop support for Python 3.4?

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -12,12 +12,6 @@ environment:
     - python: py27-x64
       tox_env: py27
       python_path: c:\python27-x64
-    - python: py34
-      tox_env: py34
-      python_path: c:\python34
-    - python: py34-x64
-      tox_env: py34
-      python_path: c:\python34-x64
     - python: py35
       tox_env: py35
       python_path: c:\python35

--- a/tox.ini
+++ b/tox.ini
@@ -1,13 +1,10 @@
 [tox]
-envlist = py{37,36,35,34,27,py3,py}
+envlist = py{37,36,35,27,py3,py}
 
 [testenv]
 deps =
     py{37,36,35,27,py3,py}: pytest-cov
     py{37,36,35,27,py3,py}: pytest-timeout
-    py{34}: pytest-timeout<1.2.1
-    py{34}: pytest<3.3
-    py{34}: pytest-cov==2.6.0
 extras = watchmedo
 commands =
     python -bb -m pytest {posargs}

